app/controllers/admin/routes.py

```diff
@@ -174,3 +175,28 @@ def addParticipants():
 
     return render_template('addParticipants.html',
                             title="Add Participants")
+
+@admin_bp.route('/courseManagement', methods = ['GET', 'POST'])
+@admin_bp.route('/<termId>/courseManagement', methods = ['GET', 'POST'])
+def courseManagement(termId = None):
+    '''
+    Renders the page for admins to manage Course Proposals
+    '''
+    currentTerm = Term.get(Term.isCurrentTerm)
+    if termId == None:
+        termId = currentTerm
+    else:
+        termId = (int(termId))
```

Comment:
you should validate termId by getting a Term object at the beginning of the controller. Then you can use that Term object later